docs/pages/includes/database-access/tctl-auth-sign-3-files.mdx

```diff
@@ -0,0 +1,60 @@
+{{ format="db" }}
+Teleport uses mutual TLS authentication with self-hosted databases. These
+databases must be able to verify certificates presented by the Teleport Database
+Service. Self-hosted databases also need a certificate/key pair that Teleport
+can verify.
+
+By default, the Teleport Database Service trusts certificates issued by a
+certificate authority managed by the Teleport Auth Service. You can either:
+- Configure your self-hosted database to trust this CA, and instruct Teleport to
+  issue a certificate for the database to present to the Teleport Database
+  Service.
+- Configure the Database Service to trust a custom CA.
+
+To configure the database to trust the Teleport CA and issue a certificate for
+the database, follow these instructions on your workstation:
+
+1. To use `tctl` from your workstation, your Teleport user must be allowed to
+   impersonate the system role `Db` in order to be able to generate the database
+   certificate. Include the following `allow` rule in in your Teleport user's
+   role:
+
+   ```yaml
+   allow:
+     impersonate:
+       users: ["Db"]
+       roles: ["Db"]
+   ```
+
+1. Export Teleport's certificate authority and a generate certificate/key pair.
+   This example generates a certificate with a 1-year validity period.
+   `db.example.com` is the hostname where the Teleport Database Service can
+   reach the {{ dbname }} server.
+
+   ```code
+   $ tctl auth sign --format={{ format }} --host=db.example.com --out=server --ttl=2190h
+   ```
+
+   (!docs/pages/includes/database-access/ttl-note.mdx!)
+
+   The command creates 3 files: `server.cas`, `server.crt` and `server.key`.
+
+If the {{ dbname }} database already has a CA that it uses to sign certificates
+, you only need to export a Teleport CA certificate for the database to
+authenticate traffic from the Teleport Database Service. You do not need to
+enable `Db` impersonation privileges.
+
+1. Replace <Var name="example.teleport.sh:443" /> with the host and web port of
+   the Teleport Proxy Service in your cluster. Run the following command on your
+   workstation:
+
+   ```code
+   $ tctl auth export --type=db-client --auth-server=<Var name="example.teleport.sh:443" /> > server.cas
+   ```
+
+   The command creates 1 file, `server.cas`.
+
+1. Generate `server.crt` and `server.key` by retrieving a TLS certificate and
+   private key from your existing database CA, signed for your database server.
+   You will use these files later in the guide.
```

Comment:
I feel in most scenarios that people want to use custom CAs, their PostgreSQL servers are already running with their own custom CA signed certs. In that case, we just need to append the db client CA in the previous step.

If they hasn't configured TLS on their PostgreSQL server yet, it would be easier they go option 1 to use Teleport's DB server CAs.

In case users want to use their own CA but hasn't configured the database servers yet (especially the cluster setups), @GavinFrazar once pointed out that the users may not know how to generate the self-signed certs. We might want to provide openssl instructions in that case, in the future. Or maybe a docker-compose file or how to use helm charts to deploy the cluster setup.

Reply:
> I feel in most scenarios that people want to use custom CAs, their PostgreSQL servers are already running with their own custom CA signed certs. In that case, we just need to append the db client CA in the previous step.

👍 Do the changes in bba4cd523e132035bfdaa9f6ed70e433e0a223c1 take care of this?

> In case users want to use their own CA but hasn't configured the database servers yet (especially the cluster setups), @GavinFrazar once pointed out that the users may not know how to generate the self-signed certs. We might want to provide openssl instructions in that case, in the future. Or maybe a docker-compose file or how to use helm charts to deploy the cluster setup.

Since this PR has been open for a while, would it make sense to add any example instructions in a separate project?

Reply:
thanks!